### PR TITLE
Contents

### PR DESCRIPTION
**Pull request by knalin55**

### CODE REVIEW

File: data/make_parquet.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+parser.add_argument("--language", default="en", type=str, help="Language (en/ru)")
+parser.add_argument("--directory", default="", type=str, help="directory where wiki dump is located")
+parser.add_argument("--num_lines", default=500000, type=int, help="Number of lines to keep in dataset")
+
+def main(args: argparse.Namespace) -> None:
+
+
+# Write first args.num_lines to a csv file. 
+    count =0
+    with open('{}.csv'.format(args.language), "w") as file:
```
Comment (jirka-x1) [inline data/make_parquet.py]: The f-strings are now typically preferred over using `format`.
```suggestion
    with open(f'{args.language}.csv', "w") as file:
```

File: data/make_parquet.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        with open(os.path.join(args.directory, '{}.txt'.format(args.language))) as r_file:
+            file.write("SNo\tinput\n")
+            for line in r_file:
+                if line.strip("\n").strip() != "":
+                    file.write(str(count) + "\t" + line.strip("\n") + " \n")
```
Comment (jirka-x1) [inline data/make_parquet.py]: Either make sure `line` does not contains `\t` or (preferably) use `csvreader` (with \t as the separator)

File: data/make_parquet.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--language", default="en", type=str, help="Language (en/ru)")
+parser.add_argument("--directory", default="", type=str, help="directory where wiki dump is located")
+parser.add_argument("--num_lines", default=500000, type=int, help="Number of lines to keep in dataset")
+
+def main(args: argparse.Namespace) -> None:
+
+
+# Write first args.num_lines to a csv file.
```
Comment (jirka-x1) [inline data/make_parquet.py]: Is it csv or tsv? And you use txt ending.

File: evaluate.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+import argparse
+import re
+from nltk.translate.bleu_score import corpus_bleu, SmoothingFunction
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--pretrain", default=None, type=str, help="Pretrained model")
+parser.add_argument("--finetune", default=None, type=str, help="Number of lines to keep in dataset")
+parser.add_argument("--ckpt", default=None, type=str, help="Number of lines to keep in dataset")
+
+def main(args: argparse.Namespace) -> None:
```
Comment (jirka-x1) [inline evaluate.py:22]: split into well defined functions (here and all other places)

File: evaluate.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        obj = "<O>"
+        pre = "<P>"
+
+
+        for entry in entry_list:
+            
+            label = ""
+            inputs = [sent.lex for i, sent in enumerate(entry.lexs) if i%2==1] if lang == "ru" else [sent.lex for i, sent in enumerate(entry.lexs)]
+            for triple in entry.list_triples():
+
```
Comment (jirka-x1) [inline evaluate.py]: In several places, you have these random empty lines. Follow PEP8. Doesn't your IDE complain?

File: evaluate.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        def camel_case_split(str):
+            return [re.findall(r"[a-z]*", str)[0]] + re.findall(r'[A-Z](?:[a-z]+|[A-Z]*(?=[A-Z]|$))', str) 
+
+        b = Benchmark()
+        files = select_files(fil_dir)
+        b.fill_benchmark(files)
+            
+        entry_list = [entry for entry in b.entries]
+        triples_list    = []
```
Comment (jirka-x1) [inline evaluate.py]: You do not explain what are triples.

File: evaluate.py [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+
+        for entry in entry_list:
+            
+            label = ""
+            inputs = [sent.lex for i, sent in enumerate(entry.lexs) if i%2==1] if lang == "ru" else [sent.lex for i, sent in enumerate(entry.lexs)]
+            for triple in entry.list_triples():
+                
+                
+                label = label + sub + " ".join(triple.split("|")[0].strip().split("_")) \
```
Comment (jirka-x1) [inline evaluate.py]: this is too hard to read; add a comment.
Also you could create a (commented) function for doing
```
def some_intuitive_name(text: str, some_intuitive_name_for_triple_index: int) -> str:
    return text.split("|")[some_intuitive_name_for_triple_index].strip().split("_")
```

File: pretrain.py [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+
+        # BT5_WMT/MT5_WMT: Training T5/MT5 further on WMT dataset. 
+        # load and process tsv file for WMT News dataset
+
+        dataset = open("./data/news-commentary-v14.en-ru.tsv").read().split("\n")
+        en = []
+        ru = []
+        for line in dataset:
+            en.append(line.split("\t")[0])
+            ru.append(line.split("\t")[1])
```
Comment (jirka-x1) [inline pretrain.py]: why not:
```suggestion
        en = [line.split("\t")[0] for line in dataset]
        ru = [line.split("\t")[1] for line in dataset]
```
But again, you should create a function for getting n-th column from a dataset.

File: finetune.py [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+
+    if args.ckpt is not None:
+        trainer.train(resume_from_checkpoint=args.ckpt)
+    else:
+        trainer.train()
+
+
+if __name__ == "__main__":
+    args = parser.parse_args([] if "__file__" not in globals() else None)
+    main(args)
```
Comment (jirka-x1) [inline finetune.py]: Make sure all files contain `\n` as the last character (See 4.6 https://ufal.mff.cuni.cz/courses/npfl096-2014-su/project)

File: pretrain.py [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+parser.add_argument("--weight_decay", default=0.01, type=float, help="Weight Decay.")
+parser.add_argument("--steps", default=200000, type=int, help="Number of steps.")
+parser.add_argument("--lr", default=2e-5, type=float, help="Learning Rate.")
+parser.add_argument("--model", default="t5-small", type=str, help="Seq to Seq Model")
+parser.add_argument("--seed", default=42, type=int, help="Random seed.")
+parser.add_argument("--threads", default=1, type=int, help="Maximum number of threads to use.")
+parser.add_argument("--ckpt_path", default=None, type=str, help="Checkpoint Path to resume from")
+
+
+def main(args: argparse.Namespace) -> None:
```
Comment (jirka-x1) [inline pretrain.py:39]: split into functions

**Review by jirka-x1 — CHANGES_REQUESTED** [review]

File: evaluate.py (hunk 5 above)
Comment (knalin55) [reply evaluate.py]: My MS VS Code had a different linter selected as default. So it didn't show the coding style-related issues. I have fixed the scripts now. It should be better than before.

File: data/make_parquet.py [hunk 11]
```diff
@@ -1,31 +1,41 @@
...
 
-# Write first args.num_lines to a csv file. 
-    count =0
-    with open('{}.csv'.format(args.language), "w") as file:
+def main(args: argparse.Namespace) -> None:
+    # Write first args.num_lines to a tsv file.
+    count = 0
+    with open(f'{args.language}.tsv', "w") as file:
+    # write n_lines to (en/ru).tsv (tmp)
         with open(os.path.join(args.directory, '{}.txt'.format(args.language))) as r_file:
```
Comment (jirka-x1) [inline data/make_parquet.py:23]: You changed the formatting to "f" string above (for the output file), but not here.

File: data/make_parquet.py [hunk 12]
```diff
@@ -1,31 +1,41 @@
...
+                if line.strip() != "":
+                    csvwriter.writerow([str(count), line.strip()])
                     count += 1
                 if count == args.num_lines:
                     break
 
-    df = pd.read_csv('{}.csv'.format(args.language), sep="\t")
-    df.to_parquet('{}.parquet'.format(args.language)) # Convert csv to parquet
-    os.remove("{}.csv".format(args.language)) # Remove intermediate file
+    df = pd.read_csv('{}.tsv'.format(args.language), sep="\t")
```
Comment (jirka-x1) [inline data/make_parquet.py:34]: ditto (and elsewhere)

File: evaluate.py [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+def main(args: argparse.Namespace) -> None:
+    langs = ["en", "ru"]
+    if args.finetune == "monolingual_ru":
+        langs = ["ru"]
+    elif args.finetune == "monolingual_en":
+        langs = ["en"]
+
+    prefix = dict(en="Generate in English ", ru="Generate in Russian ")
+
+    def extract_component(triple: str, entity_index: int) -> str:
```
Comment (jirka-x1) [inline evaluate.py:31]: why is this a function under main? (the disadvantage is that you can access any variables in main (langs, args, prefix). 
Make it a normal function (and similarly the other ones)

File: evaluate.py [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+        return triple.split("|")[entity_index].strip().split("_")
+
+    def camel_case_split(str):
+        return [re.findall(r"[a-z]*", str)[0]] +\
+            re.findall(r'[A-Z](?:[a-z]+|[A-Z]*(?=[A-Z]|$))', str)
+
+    def preprocess_data(lang):
```
Comment (jirka-x1) [inline evaluate.py:38]: what is the type of lang? All basic function arguments should have a type hint (exceptions are complex objects from third party untyped libraries, where it is not easy to find out what the type is)

File: pretrain.py [hunk 15]
```diff
@@ -0,0 +1,182 @@
...
+            for start_idx in range(0, len(dataset), 1000):
+                samples = dataset[start_idx: start_idx + 1000]
+                yield samples["input"]
+
+    elif args.pretrain_type == "bt5_wmt" or args.pretrain_type == "mt5_wmt":
+
+        # BT5_WMT/MT5_WMT: Training T5/MT5 further on WMT dataset.
+        # load and process tsv file for WMT News dataset
+
+        dataset = open("./data/news-commentary-v14.en-ru.tsv").read().split("\n")
```
Comment (jirka-x1) [inline pretrain.py:71]: yes, but obviously using a CsvReader (with \t as separator) is recommended.

**Review by jirka-x1 — CHANGES_REQUESTED** [review]

**Pull request merged**